wallets/endpoint.go

```diff
@@ -0,0 +1,161 @@
+// Copyright (C) 2022 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package wallets
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/zeebo/errs"
+	"go.uber.org/zap"
+	"storj.io/storjscan/tokens"
+)
+
+// ErrEndpoint is the wallets endpoint error class.
+var ErrEndpoint = errs.Class("Wallets Endpoint")
+
+// Endpoint for interacting with the Wallets service.
+//
+// architecture: Endpoint
+type Endpoint struct {
+	log     *zap.Logger
+	service Wallets
+}
+
+// NewEndpoint creates new wallets endpoint instance.
+func NewEndpoint(log *zap.Logger, wallets Wallets) *Endpoint {
+	return &Endpoint{
+		log:     log,
+		service: wallets,
+	}
+}
+
+// Register registers endpoint methods on API server subroute.
+func (endpoint *Endpoint) Register(router *mux.Router) {
+	router.HandleFunc("/wallets", endpoint.GetNewDepositAddress).Methods(http.MethodGet)
+	router.HandleFunc("/wallets/count", endpoint.GetCountDepositAddresses).Methods(http.MethodGet)
+	router.HandleFunc("/wallets/count/claimed", endpoint.GetCountClaimedDepositAddresses).Methods(http.MethodGet)
+	router.HandleFunc("/wallets/count/unclaimed", endpoint.GetCountUnclaimedDepositAddresses).Methods(http.MethodGet)
+	router.HandleFunc("/wallets/{address}", endpoint.GetAccount).Methods(http.MethodGet)
+}
+
+// GetNewDepositAddress returns a new deposit address.
+func (endpoint *Endpoint) GetNewDepositAddress(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	var err error
+	defer mon.Task()(&ctx)(&err)
+
+	address, err := endpoint.service.GetNewDepositAddress(ctx)
+	if err != nil {
+		endpoint.serveJSONError(w, http.StatusBadRequest, ErrEndpoint.Wrap(err))
+		return
+	}
+
+	err = json.NewEncoder(w).Encode(address)
```

Comment:
When `GetNewDepositAddress` is called, the wallet is claimed in the database. Therefore, if any issues happen after that point in which the client is unable to get the wallet returned, either with an error decoding here, or some error in returning to the caller, what happens to the wallet address in the database?

Reply:
great point... maybe there should be an endpoint to get a new address, then another to claim it when the user confirms? Right now you're right, it would just be marked claimed regardless of whether the user actually receives the address/the satellite records it as tied to the user.

Reply:
resolution: it's ok if network errors cause claimed addresses in the db for now.